tests/testthat/test-sptf-bulkdensity.R

```diff
@@ -1,10 +1,234 @@
 # test bulk density functions
 require(testthat)
 
-test_that("estimation sptf_bd1", {
-  expect_equal(
-    sptf_bd1(A_SOM_LOI = c(2,5), 
-             A_CLAY_MI = c(10,15)),
-    expected = c(1343.75,1176.612),
-    tolerance = 0.01)
+# dynamic file location
+tloc <- if(length(list.files("../testdata/"))>0){"../testdata/"} else {'tests/testdata/'}
+
+# test for all functions of bulk density
+# if not passing test it will indicate what collumn (function) name is not right
+
+
+# define inputs
+dt <- data.table(
+  A_SOM_LOI = c(2,5),
+  A_C_OF = c(20, 200),
+  A_CLAY_MI = c(10, 20),
+  A_SILT_MI = c(23, 10),
+  A_SAND_MI = c(15, 25),
```

Comment:
Sand clay and silt do not add up to 100 which may create some unrealistic test cases. A function that uses these three really should check this by itself and throw an error

Reply:
yes indeed. [issue](https://github.com/AgroCares/soilptf/issues/14)